src/app/dataset-view/additional-components/dataset-settings-component/tabs/access/dataset-settings-access-tab/dataset-settings-access-tab.component.ts

```diff
@@ -0,0 +1,207 @@
+import {
+    AccountWithRole,
+    DatasetAccessRole,
+    DatasetBasicsFragment,
+    PageBasedInfo,
+} from "src/app/api/kamu.graphql.interface";
+import { DatasetViewTypeEnum } from "src/app/dataset-view/dataset-view.interface";
+import { SettingsTabsEnum } from "../../../dataset-settings.model";
+import { MatTableDataSource } from "@angular/material/table";
+import { BaseComponent } from "src/app/common/components/base.component";
+import { promiseWithCatch, requireValue } from "src/app/common/helpers/app.helpers";
+import ProjectLinks from "src/app/project-links";
+import { NavigationService } from "src/app/services/navigation.service";
+import { NgbModal } from "@ng-bootstrap/ng-bootstrap";
+import { AddPeopleModalComponent } from "./add-people-modal/add-people-modal.component";
+import { ModalService } from "src/app/common/components/modal/modal.service";
+import { SelectionModel } from "@angular/cdk/collections";
+import { DatasetCollaborationsService } from "./dataset-collaborations.service";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import { filter, from } from "rxjs";
+import { NavigationEnd, Router } from "@angular/router";
+
+@Component({
+    selector: "app-dataset-settings-access-tab",
+    templateUrl: "./dataset-settings-access-tab.component.html",
+    styleUrls: ["./dataset-settings-access-tab.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DatasetSettingsAccessTabComponent extends BaseComponent implements OnInit {
+    @Input({ required: true }) public datasetBasics: DatasetBasicsFragment;
+
+    public dataSource = new MatTableDataSource();
+    public currentPage = 1;
+    public pageBasedInfo: PageBasedInfo;
+    public readonly PER_PAGE = 15;
+    public searchMember = "";
+    public selection = new SelectionModel<AccountWithRole>(true, []);
+
+    public readonly DISPLAY_COLUMNS: string[] = ["user", "role", "actions"];
+    public readonly DatasetViewTypeEnum: typeof DatasetViewTypeEnum = DatasetViewTypeEnum;
+    public readonly SettingsTabsEnum: typeof SettingsTabsEnum = SettingsTabsEnum;
+
+    private navigationService = inject(NavigationService);
+    private ngbModalService = inject(NgbModal);
+    private modalService = inject(ModalService);
+    private datasetCollaborationsService = inject(DatasetCollaborationsService);
+    private cdr = inject(ChangeDetectorRef);
+    private router = inject(Router);
+
+    public get isPrivate(): boolean {
+        return this.datasetBasics.visibility.__typename === "PrivateDatasetVisibility";
+    }
+
+    public ngOnInit(): void {
+        this.getPageFromUrl();
+        this.updateTable(this.currentPage);
+        this.router.events
+            .pipe(filter((event) => event instanceof NavigationEnd))
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe(() => {
+                this.getDatasetInfoFromUrl();
+                this.updateTable(this.currentPage);
+            });
```

Comment:
Similarly, we've just learned how to map those changes to @Input. For changes, there is `ngOnChanges` callback.

Reply:
The same